app/controllers/projects/samples_controller.rb

```diff
@@ -183,5 +223,30 @@ def set_metadata_fields
         show_fields: params[:q] && params[:q][:metadata].to_i == 1
       )
     end
+
+    def destroy_multiple_params
+      params.require(:multiple_deletion).permit(sample_ids: [])
+    end
+
+    def delete_multiple_samples(samples)
```

Comment:
I think it would be better for us to create a new service under `samples` such as `multi_destroy_service` which will accept the project, and samples list. The service would:
- authorize if the user can destroy a sample for the project,
- validate that all samples belong to the project (remove any that aren't from the project), 
- then you can use one query to remove the sample ids that belong to the project. 

Currently it would use multiple queries per sample, one to authorize each sample.project, and then one to remove the sample

Reply:
Added in f2e63ef19558a820a4b410ade62d32ac2129a66a